### PR TITLE
Release 2.8.0: Add query filtering on role to members.listMembers

### PR DESCRIPTION
**Pull request by mtompset**
### Add
* query filtering on role to members.listMembers

### CODE REVIEW

**Review by jason-jackson — COMMENTED** [review]
Given that this is public and this change could break it for others, I think we shouldn't assume even if it is easier to code.

Granted, no one else seems to be using this, at least according to Github stats.

File: SilMock/Google/Service/Directory/Resource/Members.php [hunk 1]
```diff
@@ -69,19 +69,19 @@ public function listMembers($groupKey, $optParams = [])
         $this->validateGroupExists($groupKey);
         $pageSize = $optParams['pageSize'] ?? 10;
         $pageToken = $optParams['pageToken'] ?? 0;
+        $query = $optParams['query'] ?? null;
+        $expectedRole = $query ? substr($query, strpos($query, '=') + 1) : null;
```
[the remark this answers is not in the document]
Comment (mtompset) [reply SilMock/Google/Service/Directory/Resource/Members.php:73]: https://developers.google.com/admin-sdk/directory/v1/guides/manage-group-members#get_all_members

[the remark this answers is not in the document]
Comment (mtompset) [reply SilMock/Google/Service/Directory/Resource/Members.php]: It's new code. It's not going to break it for anyone. Fine. I'll spend the extra time to implement it better.

**Review by briskt — APPROVED** [review]
I don't understand why the query separator is a space character, but other than that it looks fine. By not supporting query parameters other than `roles`, you risk having to rewrite this code another time in the future when you're not as fresh on its inner workings. Having said that, I appreciate not wanting to invest much time in a mock.

**Comment by mtompset** [comment]
The separator is a space because ... shoot can't find the query language page..   it has ANDs and ORs and brackets.

**Comment by mtompset** [comment]
"To search multiple fields in a query, add each search clause, separated by a space. The operation is an implicit AND." (https://developers.google.com/admin-sdk/directory/v1/guides/search-users) -- same for groups, members, and customers.

**Comment by mtompset** [comment]
Also, I code what I  need. Next release will be batching.

**Pull request merged**